h/services/annotation_moderation.py

```diff
@@ -18,9 +20,30 @@ def all_hidden(self, annotation_ids):
         query = self._session.query(AnnotationModeration.annotation_id).filter(
             AnnotationModeration.annotation_id.in_(annotation_ids)
         )
-
         return {m.annotation_id for m in query}
 
+    def set_status(
+        self, annotation: Annotation, user: User, status: ModerationStatus | None
+    ) -> None:
+        """Set the moderation status for an annotation."""
+        if status and status != annotation.moderation_status:
+            self._session.add(
+                ModerationLog(
+                    annotation_id=annotation.id,
+                    old_moderation_status=annotation.moderation_status,
+                    new_moderation_status=status.value,
+                    user_id=user.id,
+                )
+            )
+            annotation.moderation_status = status
+
+    def update_status(self, annotation: Annotation) -> None:
```

Comment:
This method will grow more parameters to make status transactions, for now it just handles "initializing" status.

Reply:
I reckon this will be more clear once update_status has more parameters and handles all types of transactions.

Let check how it look on the PR that add that.